src/nspawn/nspawn.c

```diff
@@ -1104,6 +1105,12 @@ static int parse_argv(int argc, char *argv[]) {
         if (e)
                 arg_container_service_name = e;
 
+        r = getenv_bool("SYSTEMD_NSPAWN_USE_CGNS");
+        if (r < 0)
+                arg_use_cgns = cg_ns_supported();
+        else
+                arg_use_cgns = r;
```

Comment:
If the user sets `SYSTEMD_NSPAWN_USE_CGNS=1` but the kernel does not support it, nspawn will silently ignore the user request and continue without cgns. That's because, even though `arg_use_cgns` is assigned to `1` just here, `cg_ns_supported()` is called again further down.

That's different than `UNIFIED_CGROUP_HIERARCHY` semantics, where if the user sets `UNIFIED_CGROUP_HIERARCHY=1` but the kernel does not support it, nspawn will just fail and print the error.

I think I prefer `UNIFIED_CGROUP_HIERARCHY` semantics: if the user asks something, but it's not possible, it's better to error with a message rather than ignore the request...

Could we do the same for `SYSTEMD_NSPAWN_USE_CGNS`?